### PR TITLE
fix: Not working Clear button

### LINKED ISSUE #116: Clear shortcut not working
**Describe the bug**
You can set a key to NONE in the setting dialog, but right-click - clear is not working.

**Configuration**
Present in master and 3.1.3

### PR DESCRIPTION
**Pull request by pktiuk**
Closes #116

### CODE REVIEW

**Comment by pktiuk** [comment]
There is still one thing to do, refresh content of button showing assigned key.  
It will refresh anyway when you click button on gamepad, but I still think it would be good to provide instant feedback to user that this config has been cleared.

I am also concerned about commit da19b141038a77b739d4bf43fa49330e30ad9678, which introduced this bug. It points at possible memory leaks.

We will have to investigate it.

**Comment by pktiuk** [comment]
I don't know how to reach function changing label of restarted button from function cleaning assigned buttons.

**Comment by gombosg** [comment]
I tested now and without updating the labels it's not very usable. The users would still think that it hasn't been cleared. Any way to do this?

**Comment by pktiuk** [comment]
I will try to dig deeper to find what is wrong, but it may take a while.

**Comment by gombosg** [comment]
It's OK. What comes to my mind: how did it ever work before? :)

**Comment by pktiuk** [comment]
It worked in antimicro.

**Comment by pktiuk** [comment]
I have bisected this bug, it is around 8153c88c54a33901ec9751e47e0dd521fa237b67

**Comment by pktiuk** [comment]
It is 6521eed051cf0b52ba45e72a5e86d518998c109e. Unluckily this is a very big commit, and it may take some time to read it.

**Comment by pktiuk** [comment]
Fixed

File: src/joybutton.cpp [hunk 1]
```diff
@@ -3305,7 +3305,7 @@ void JoyButton::clearSlotsEventReset(bool clearSignalEmit)
     QWriteLocker tempAssignLocker(&assignmentsLock);
 
     resetSlotsProp();
-    stopTimers(true, true, false);
+    stopTimers(true, false, false);
```
Comment (pktiuk) [inline src/joybutton.cpp:3308]: This is funny, how such a small and old mistake can break many things.

**Comment by gombosg** [comment]
Wow, awesome job! 🙌 I tested, it's good now.

**Pull request merged**